### ISSUE TITLE
Can't replicate flatland reward

### ISSUE THREAD

**Issue by janetwise** [jbr-ai-labs/mamba#6]
Hello,

Thank you for sharing the mamba code. However, I couldn't replicate your 5-agent flatland reward curve as shown in your paper (Fig.5). Can you confirm the hyperparameters used to achieve that plot? 

Thank you very much in advance!

Janet

**Comment by James0618** [comment]
Hi, I have the same question. The performance is so poor!!!

**Comment by vladimirrim** [comment]
Hi,

did you copy optimal hyperparameters from `optimal/flatland` to [DreamerAgentConfig.py](https://github.com/jbr-ai-labs/mamba/blob/main/configs/dreamer/DreamerAgentConfig.py) and [DreamerLearnerConfig.py](https://github.com/jbr-ai-labs/mamba/blob/main/configs/dreamer/DreamerLearnerConfig.py)?

I did rerun an experiment from scratch with `python3 train.py` and it's learning as expected -- wandb [link](https://api.wandb.ai/links/vladimirrim/ddua3o1f) (Note that x-axis is different from the number of steps in the enviornment)

**Comment by janetwise** [comment]
Thank you so much, I just used those default files but not copied from the optimal folder. in the will give it a try.

How many world models did you use? The paper says each agent has their own version of the world model, I couldn't find it in the code. The code seems to indicate that only a single world model is learned with parameter sharing, am I understanding it right? Both policy/actor and critic also use/train just one model with shared parameters across all agents. Is it right? 

Also, what exactly does the communication block learn? 

Thanks in advance!

**Comment by vladimirrim** [comment]
> The code seems to indicate that only a single world model is learned with parameter sharing, am I understanding it right? Both policy/actor and critic also use/train just one model with shared parameters across all agents. Is it right?

Yes, your understanding is correct. Agents can have their own copy of the same model in theory, but it will reduce the training speed.

> Also, what exactly does the communication block learn?

Communication block can be seen as the attention layer in [Transition](https://github.com/jbr-ai-labs/mamba/blob/2972a1ed1e139a19a661cd506d7f4ca5e9cac856/networks/dreamer/rnns.py#L39) model, which extract information needed to predict the next state from previous actions and states of agents

Hope this helps clarify the architecture
